src/vehicle.py

```diff
@@ -40,42 +39,64 @@ def __init__(self, player: Player, sprite_lists: SpriteLists):
             RocketLauncher,
             MachineGun,
         ]
-        self.weapon_index = 0
+        self.weapon_index: int = 0
         self.primary_weapon: Weapon
         self.secondary_weapon: Weapon
         self.primary_weapon_sprite: arcade.Sprite
         self.secondary_weapon_sprite: arcade.Sprite
         self._swap_in_weapons()
-        self.x_shift = float
-        self.y_shift = float
+        self.hit_indicator: bool = False
+        self.time_since_hit: float = 0
+        self.fire_sprite = arcade.Sprite(texture=FIRE, scale=3)
         self.drive_mode_index = 0
         self.drive_modes = create_drive_modes(self)
-        self.velocity = (0.0, 0.0)
-        "Translational velocity -- (x,y) tuple -- measured in pixels per second"
 
         self.movement = MovementControls(
             LinkedSprite[Vehicle](texture=RED_CAR, scale=0.18)
         )
 
     def update(self, delta_time: float):
-        #
+
         # Driving and movement
-        #
         if self.player.alive:
             self.movement.drive_input(delta_time, self, self.player.input)
         self.movement.move(delta_time, self, self.sprite_lists.walls)
 
-        #
         # Weapons
-        #
         if self.player.alive:
+            self.movement.drive_input(delta_time, self, self.player.input)
             self.primary_weapon.update(delta_time)
             self.secondary_weapon.update(delta_time)
             if self.player.input.swap_weapons_button.pressed:
                 self._swap_weapons()
 
+        if self.hit_indicator:
+            self.time_since_hit += delta_time
+            if self.time_since_hit > 0.2:
```

Comment:
Nit, non-blocking: magic number, could be moved to a constant at the top of this file, like `HIT_INDICATOR_DURATION = 0.2`

Reply:
Implimented